central/image/service/http_handler.go

```diff
@@ -0,0 +1,153 @@
+package service
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/pkg/errors"
+	clusterUtil "github.com/stackrox/rox/central/cluster/util"
+	"github.com/stackrox/rox/central/role/sachelper"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/features"
+	"github.com/stackrox/rox/pkg/httputil"
+	"github.com/stackrox/rox/pkg/images/enricher"
+	"github.com/stackrox/rox/pkg/images/integration"
+	"google.golang.org/grpc/codes"
+)
+
+const (
+	NumberBytes = 100 * 1024
+	timeout     = 10 * time.Minute
+)
+
+type sbomRequestBody struct {
+	Cluster   string `json:"cluster"`
+	ImageName string `json:"image_name"`
+	Force     bool   `json:"force"`
+}
+
+type sbomHttpHandler struct {
+	integration      integration.Set
+	enricher         enricher.ImageEnricher
+	clusterSACHelper sachelper.ClusterSacHelper
+}
+
+var _ http.Handler = (*sbomHttpHandler)(nil)
+
+// SBOMHandler returns a handler for get sbom http request
+func SBOMHandler(integration integration.Set, enricher enricher.ImageEnricher, clusterSACHelper sachelper.ClusterSacHelper) http.Handler {
+	return sbomHttpHandler{
+		integration:      integration,
+		enricher:         enricher,
+		clusterSACHelper: clusterSACHelper,
+	}
+
+}
+
+func (h sbomHttpHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+
+	if r.Method != http.MethodPost {
+		w.WriteHeader(http.StatusMethodNotAllowed)
+		return
+	}
+	// verify scanner v4 is enabled
+	if !features.ScannerV4.Enabled() {
+		httputil.WriteGRPCStyleError(w, codes.NotFound, errors.New("Scanner v4 is not enabled. Enabled scanner v4 to get SBOMs"))
+		return
+	}
+	if !features.SBOMGeneration.Enabled() {
+		httputil.WriteGRPCStyleError(w, codes.NotFound, errors.New("SBOM feature is not enabled"))
```

Comment:
nit:
Returning `codes.NotFound` for these two feature checks may not be the best fit because the endpoint / resource was found but the server is in a state where the request cannot be processed.

Because it's the server 'issue', maybe one of these would be a better fit?

1. [codes.Internal](https://github.com/grpc/grpc-go/blob/v1.68.1/codes/codes.go#L168-L174) ([HTTP `500`](https://github.com/grpc-ecosystem/grpc-gateway/blob/v2.22.0/runtime/errors.go#L67-L68))
1. [codes.Unimplemented](https://github.com/grpc/grpc-go/blob/v1.68.1/codes/codes.go#L158-L166) ([HTTP `501`](https://github.com/grpc-ecosystem/grpc-gateway/blob/v2.22.0/runtime/errors.go#L69-L70))

Reply:
done